### PR TITLE
[CSL-1944] Document `Pos.Wallet.Web.State.Storage`

### PR DESCRIPTION
**Pull request by flyingleafe**
Provided doc comments for `Pos.Wallet.Web.State.Storage`

PR is WIP, because `Acidic` and `State` modules should be documented too (though they mostly consist of boilerplate code, so there's less work to do).

Also during documenting I realized that there's _lots_ of old code which is now obsolete in `Pos.Wallet.Web.State.Storage`. I'd like to remove this code instead of documenting it. @adinapoli-iohk what do you think about it?

### CODE REVIEW

File: wallet/src/Pos/Wallet/Web/State/Storage.hs [hunk 1]
```diff
@@ -328,27 +428,34 @@ removeCustomAddress t (addr, hh) = do
...
     whenJustM (preuse accInfo) $ \info -> do
         let mAddr = info ^. aiAddresses . at cwamId
         when (isNothing mAddr) $ do
+            -- Here we increment current account's last address index
+            -- and assign its value to sorting index of newly created address.
             accInfo . aiUnusedKey += 1
             let key = info ^. aiUnusedKey
             accInfo . aiAddresses . at cwamId ?= AddressInfo addrMeta key
 
+-- | TODO: what happens here?...
```
Comment (akegalj) [inline wallet/src/Pos/Wallet/Web/State/Storage.hs]: Should this be addressed?

File: wallet/src/Pos/Wallet/Web/State/Storage.hs [hunk 2]
```diff
@@ -359,59 +466,76 @@ addRemovedAccount addrMeta@CWAddressMeta{..} = do
...
 addWalletTxHistory cWalId cTxId cTxMeta =
     wsTxHistory . at cWalId . non' _Empty . at cTxId ?= cTxMeta
 
+-- | Update meta data for multiple wallet transactions at once.
 setWalletTxHistory :: CId Wal -> [(CTxId, CTxMeta)] -> Update ()
 setWalletTxHistory cWalId cTxs = mapM_ (uncurry $ addWalletTxHistory cWalId) cTxs
 
--- FIXME: this will be removed later (temporary solution)
+-- | Set meta data for transaction only if it hasn't been set already.
+-- FIXME: this will be removed later (temporary solution) (not really =\)
```
Comment (akegalj) [inline wallet/src/Pos/Wallet/Web/State/Storage.hs]: :/

File: wallet/src/Pos/Wallet/Web/State/Storage.hs (hunk 1 above)
Comment (flyingleafe) [reply wallet/src/Pos/Wallet/Web/State/Storage.hs]: Yes. I'm genuinely confused about this function.

**Comment by adinapoli-iohk** [comment]
>  I'd like to remove this code instead of documenting it. @adinapoli-iohk what do you think about it?

Absolutely, removed code is the best 😉

File: wallet/src/Pos/Wallet/Web/State/Storage.hs [hunk 3]
```diff
@@ -110,59 +110,113 @@ import           Pos.Wallet.Web.Pending.Types (PendingTx (..), PtxCondition, Ptx
...
+    { -- | Wallet metadata (name, assurance, etc.)
+      _wiMeta         :: !CWalletMeta
+      -- | Last time when wallet passphrase was updated.
     , _wiPassphraseLU :: !PassPhraseLU
+      -- | Wallet creation time.
     , _wiCreationTime :: !POSIXTime
+      -- | Header hash until which wallet has been synced
+      -- (or @NotSynced@ if wallet hasn't beed synced at all)
     , _wiSyncTip      :: !WalletTip
+      -- | Set of transactions which are pending for resubmission
```
Comment (Martoon-00) [inline wallet/src/Pos/Wallet/Web/State/Storage.hs]: It's not exactly true. We keep this "pending state" for all transactions made

File: wallet/src/Pos/Wallet/Web/State/Storage.hs [hunk 4]
```diff
@@ -359,59 +466,76 @@ addRemovedAccount addrMeta@CWAddressMeta{..} = do
...
 
+-- | Delete all transactions' metadata for given wallet.
 removeTxMetas :: CId Wal -> Update ()
 removeTxMetas cWalId = wsTxHistory . at cWalId .= Nothing
 
+-- | 'addOnlyNewTxMeta' for several transactions at once.
 addOnlyNewTxMetas :: CId Wal -> [(CTxId, CTxMeta)] -> Update ()
 addOnlyNewTxMetas = mapM_ . uncurry . addOnlyNewTxMeta
 
+-- | Delete metadata by given wallet id and transactions ids.
```
Comment (Martoon-00) [inline wallet/src/Pos/Wallet/Web/State/Storage.hs]: Since you mention `wallet` here, maybe clarify: `Delete transaction metadata`?

**Comment by Martoon-00** [comment]
It was so painful so that you failed to document up to the end? 😄

**Comment by Martoon-00** [comment]
Ah ok I see, let's cleanup first 👍

File: wallet/src/Pos/Wallet/Web/State/Storage.hs (hunk 1 above)
Comment (Martoon-00) [reply wallet/src/Pos/Wallet/Web/State/Storage.hs]: Indeed, it's not used anywhere :peka:

File: wallet/src/Pos/Wallet/Web/State/Storage.hs (hunk 2 above)
Comment (Martoon-00) [reply wallet/src/Pos/Wallet/Web/State/Storage.hs]: _I'm not going to see anime with drama anymore_
_I have enough of it in code_ :droplet:

**Comment by flyingleafe** [comment]
I think that now this PR is enough, please review.

File: wallet/src/Pos/Wallet/Web/State/Storage.hs [hunk 5]
```diff
@@ -423,34 +511,35 @@ removeWAddress addrMeta@(addrMetaToAccount -> accId) = do
...
+-- | Remove next pending update info from update info list.
 removeNextUpdate :: Update ()
 removeNextUpdate = wsReadyUpdates %= drop 1
 
+-- | Reset the whole database to clean state completely. Used only in testing and debugging.
 testReset :: Update ()
 testReset = put def
 
--- Legacy transaction, no longer used. For existing Db tx logs only. Now use
+-- | Legacy transaction, no longer used. For existing Db tx logs only. Now use
```
Comment (Martoon-00) [inline wallet/src/Pos/Wallet/Web/State/Storage.hs]: Minor: how about removing this method as well?

Comment (Martoon-00) [reply wallet/src/Pos/Wallet/Web/State/Storage.hs]: Ah okay, I see that you doesn't do migration

**Pull request merged**